### PR TITLE
testbed: Support UDP in addition to TCP for agent health check

### PR DESCRIPTION
**Pull request by WPH95**
**Description:**
Testbed hardcode **"tcp"** in `agent_start/health_check`. but I need add testbed add a syslog_input with **"udp"** testcase.

so DataSender interface add a `GetNetworkProtocol() string`, default is "tcp". make `agent_start/health_check` support udp

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2528

### CODE REVIEW

File: testbed/testbed/senders.go [hunk 1]
```diff
@@ -54,6 +54,9 @@ type DataSender interface {
 	// Return the port to which this sender will send data.
 	GetEndpoint() string
 
+	// Return the network protocol which this sender will send data can access
```
Comment (tigrannajaryan) [inline testbed/testbed/senders.go:57]:
```suggestion
	// Return the network protocol which this sender will send data. One of "tcp" or "udp".
	// GetEndpoint() and GetNetworkProtocol() together can used for probing whether the recipient
	// is ready to accept data (see TestCase.StartAgent() implementation).
```

**Review by tigrannajaryan — APPROVED** [review]

File: testbed/testbed/senders.go [hunk 2]
```diff
@@ -114,6 +119,10 @@ func (dsb *DataSenderBase) Flush() {
 	// Exporter interface does not support Flush, so nothing to do.
 }
 
+func (dsb *DataSenderBase) GetNetworkProtocol() string {
```
Comment (bogdandrutu) [inline testbed/testbed/senders.go:122]: Is it not better to change `GetEndpoint` to return `net.IpAddr`?

Comment (WPH95) [reply testbed/testbed/senders.go:122]: need use testbed test a udp server, use `net.Ipaddr` no way to distinguish between tcp and udp

Comment (bogdandrutu) [reply testbed/testbed/senders.go:122]: Sorry, I meant `net.Addr` which has a network value which can be tcp/udp/etc.

Comment (WPH95) [reply testbed/testbed/senders.go:122]: good idea, let me change it :)

**Comment by WPH95** [comment]
still have some strange ci test failed (markdown lint?) 
try to:
   used the latest master branch
   git patch this pr change
   git push force

**Comment by bogdandrutu** [comment]
Please rebase, will pass links tests

**Pull request merged**